src/Countries/India.php

```diff
@@ -0,0 +1,475 @@
+        ];
+    }
+
+    /**
+     * In India there are more state specific holidays in compared to National Holidays
+     * We can implement a mechanism to check the state and return the holidays accordingly
+     */
+    protected function variableHolidays(int $year, ?string $region = null): array
+    {
+        if (empty($region)) {
+            return [];
+        }
+        return match ($region) {
+            'AD', 'Andhra Pradesh' => [
+                'Ugadi' => '04-13',
+                'Sri Rama Navami' => '04-21',
+                'Buddha Purnima' => '05-26',
+                'Eid al-Fitr' => '05-14',
+                'Bonalu' => '07-11',
+                'Eid al-Adha' => '07-20',
+                'Muharram' => '08-19',
+                'Ganesh Chaturthi' => '09-10',
+                'Dussehra' => '10-15',
+                'Milad un-Nabi' => '10-19',
+                'Diwali' => '11-04',
+                'Kartika Purnima' => '11-19',
+            ],
+            'AR', 'Arunachal Pradesh' => [
+                'Losar' => '02-12',
+                'Nyokum Yullo' => '02-26',
+                'Mopin' => '04-05',
+                'Tamladu' => '08-15',
+                'Solung' => '09-01',
+                'Dree' => '10-04',
+                'Karnataka Rajyotsava' => '11-01',
+                'Pangsau Pass Winter Festival' => '12-20',
+            ],
+            'AS', 'Assam' => [
+                'Magh Bihu' => '01-15',
+                'Tusu Puja' => '01-15',
+                'Me-Dum-Me-Phi' => '01-31',
+                'Bhogali Bihu' => '01-15',
+                'Sivaratri' => '03-11',
+                'Dol Jatra' => '03-28',
+                'Bohag Bihu' => '04-14',
+                'Rongali Bihu' => '04-14',
+            ],
+            'BR', 'Bihar' => [
+                'Makar Sankranti' => '01-14',
+                'Vasant Panchami' => '02-16',
+                'Maha Shivaratri' => '03-11',
+                'Holi' => '03-29',
+                'Good Friday' => '04-02',
+                'Ram Navami' => '04-21',
+                'Mahavir Jayanti' => '04-25',
+                'Buddha Purnima' => '05-26',
+                'Eid al-Fitr' => '05-14',
+                'Raksha Bandhan' => '08-22',
+                'Janmashtami' => '08-30',
+                'Ganesh Chaturthi' => '09-10',
+                'Muharram' => '08-19',
+                'Dussehra' => '10-15',
+                'Milad un-Nabi' => '10-19',
+                'Diwali' => '11-04',
+                'Chhath Puja' => '11-10',
+            ],
+            'CG', 'Chhattisgarh' => [
+                'Makar Sankranti' => '01-14',
+                'Vasant Panchami' => '02-16',
+                'Maha Shivaratri' => '03-11',
+                'Holi' => '03-29',
+                'Good Friday' => '04-02',
+                'Ram Navami' => '04-21',
+                'Mahavir Jayanti' => '04-25',
+                'Buddha Purnima' => '05-26',
+                'Eid al-Fitr' => '05-14',
```

Comment:
I don't think a fixed list can be used for Islamic holidays

Reply:
There we would need a manual intervention for adding Islamic and Hindu holidays based on Moon based calendar, this can either come through config or via class abstract methods